Content/Items/Misc/Accessories.BricklayersMallet.cs

```diff
@@ -0,0 +1,43 @@
+﻿using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Core;
+using Terraria.ID;
+using Terraria;
+using System;
+using Microsoft.Xna.Framework;
+using StarlightRiver.Core.Systems.ChestLootSystem;
+using Terraria.ModLoader;
+
+namespace StarlightRiver.Content.Items.Misc
+{
+    public class BricklayersMallet : SmartAccessory
+    {
+        public override string Texture => AssetDirectory.MiscItem + Name;
+        public BricklayersMallet() : base("Bricklayer's Mallet", "Doubles block placement and tool range\nDecreases mining speed by 50% for blocks outside your original range") { }
+
+        public override void SafeSetDefaults()
+        {
+            Item.rare = ItemRarityID.Green;
+            Item.value = Item.sellPrice(silver: 25);
+        }
+
+        public override void SafeUpdateEquip(Player Player)
+        {
+            Player.blockRange *= 2;
+            if (Main.myPlayer == Player.whoAmI)
+            {
+                Player.tileRangeX *= 2;
+                Player.tileRangeY *= 2;
+
+                bool outsideXRange = Main.MouseWorld.X < Player.Center.X ? Math.Abs(Player.Left.X - Main.MouseWorld.X) > 80f : Math.Abs(Player.Right.X - Main.MouseWorld.X) > 80f;
```

Comment:
Should these be hardcoded? What about other vanilla accessories that extend the default range?

Reply:
your default range still stays the same? if something increases your default, that isn't your new default

Reply:
default meaning "not affected by this accessory" here. This is currently written to override all additive effects that apply before this.